dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/generate/ServiceImplementationGenerator.java

```diff
@@ -294,13 +295,14 @@ public CodeBlock primitive(TypeName _typeName, String parameterSerializerMethodN
             }
 
             @Override
-            public CodeBlock list(TypeName _typeName, String parameterSerializerMethodName) {
+            public CodeBlock list(TypeName _typeName, ListType listType) {
                 return maybeParameterEncoderType.map(this::parameterEncoderType).orElseGet(() -> {
+                    CodeBlock elementName = CodeBlock.of("$L$L", argName, "Element");
                     CodeBlock asList = CodeBlock.of(
-                            "$L.stream().map($L::$L).collect($T.toList())",
+                            "$L.stream().map($L -> $L).collect($T.toList())",
```

Comment:
It might be better to make this a series of maps:
- It allows us to use method references which (in theory) allows to compiler to avoid allocation
- It avoids the need to define a variable here, which avoid potential name conflicts

```suggestion
                            "$L.stream()$L.collect($T.toList())",
```